include/flox/resolver/environment.hh

```diff
@@ -283,6 +282,48 @@ private:
 
 public:
 
+  void
+  setGlobalManifest( std::optional<GlobalManifest> manifest )
+  {
+    this->globalManifest = manifest;
+    this->environment    = std::nullopt;
```

Comment:
Blocker: All of the method names in this diff chunk are misleading. Why does a method called `setGlobalManifest` do _anything_ to the `environment`? I should be able to tell at the call site what this is doing, but these methods all have (surprising) side effects that aren't reflected in the method name at all.

Reply:
This function `setGlobalManifest` sets a variable named `globalManifest`.

The function name is appropriate. `getVar` and `setVar` are the single most common pattern in OOP.

Reply:
> It invalidates environment intentionally because it is now stale. It will be refreshed when accessed through other `get*` fns.

Then do that outside this function.

> This function `setGlobalManifest` sets a variable named `globalManifest`.  The function name is appropriate.

This would be true if this is all it was doing, but that's not the case. 

> "block" on defensive/pre-caution questions, or actionable change recommendations. Please avoid blocking _statements_ or _smells_.

The purpose of code review is to make sure that the submitted code is understandable and does what it says it does. This isn't a matter of a temporary hack, a partial solution, or a first iteration. Control flow in pkgdb is already hard to understand and implicit side effects will only make it harder to understand.

Reply:
Fair enough.

I went ahead and pushed some changes that will hopefully make this easier to understand.